0000-spec-distribution.md

```diff
@@ -0,0 +1,313 @@
+It allows individual buildpacks to be distributed via a Docker registry, and it makes dynamic assembly of builders on a Docker registry efficient.
+It provides a developer-friendly interface that abstracts away complex buildpack configurations.
+
+# What it is
+[what-it-is]: #what-it-is
+
+This RFC proposes an official way to distribute buildpacks that conform to the CNB buildpack v3 specification.
+Changes would consist of a new Buildpack Distribution specification, modifications to the lifecycle builder, and modifications to the pack CLI.
+
+It affects all personas that interact with buildpacks.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+## Overview of Changes
+
+- The `order.toml` and `buildpack.toml` files are merged into `buildpack.toml`.
+- `buildpack.toml` can contain multiple buildpacks.
+- A buildpack inside of `buildpack.toml` is either:
+  a. a single buildpack implementation specified by a `path` field or
+  b. an ordering of other buildpack IDs/versions
+- Every buildpack inside `buildpack.toml` must have an ID and version.
+- Multiple buildpack implementations can be defined in `buildpack.toml`, so buildpacks can share files/code.
+- A buildpackage is a collection of buildpack repositories with `buildpack.toml` files.
+- A buildpackage can have a default buildpack.
+- Labels are removed from the specification.
+- The lifecycle moves into `/cnb/lifecycle/`.
+
+## Buildpack Blob Format
+
+A buildpack blob is a tgz containing a `buildpack.toml`, which is a configuration file defined in the CNB buildpack specification.
+Example format:
+
+```toml
+[[buildpacks]]
+id = "io.buildpacks.nodejs"
+name = "Node.js Buildpack"
+version = "0.0.9"
+[[buildpacks.order]]
+group = [
+   { id = "io.buildpacks.node", version = "0.0.5" },
+   { id = "io.buildpacks.npm", version = "0.0.7" },
+]
+
+[[buildpacks]]
+id = "io.buildpacks.npm"
+name = "NPM Buildpack"
+version = "0.0.7"
+path = "./npm-cnb/"
+[buildpacks.metadata]
+# ...
+[[buildpacks.stacks]]
+id = "io.buildpacks.stacks.bionic"
+
+[[buildpacks]]
+id = "io.buildpacks.node"
+name = "Node Engine Buildpack"
+version = "0.0.5"
+path = "./node-cnb/"
+[buildpacks.metadata]
+# ...
+[[buildpacks.stacks]]
+id = "io.buildpacks.stacks.bionic"
+
+```
+
+Each `path` must reference a valid buildpack implementation.
+However, buildpacks defined in `[[buildpacks.order]]` do not need to be included in the buildpack blob.
+
+So that a buildpack can identify its corresponding entry in `buildpack.toml`, the buildpack ID is set during the detection and build phases in `$BP_ID`.
+
+## Buildpackage Format
+
+A buildpackage may exist as an OCI image on an image registry, an OCI image in a Docker daemon, or a `.cnb` file.
+
+A `.cnb` file is an uncompressed tar archive containing an OCI image. Its file name should end in `.cnb`.
+
+### Layer Blob
+
+Each FS layer blob in the image contains a single buildpack blob tgz and at least one symlink.
```

Comment:
is the "single *buildpackage* blob" or "single tgz containing buildpacks"? Because the example below is confusing given that the checksums for each differant ID is the same. Maybe I'm just misreading here.

Reply:
A blob always means a tgz containing exactly one buildpack.toml and any number of buildpacks (either order definitions or concrete buildpacks).

Reply:
✔️ thanks.